benchmark/benchmain/main.go

```diff
@@ -648,76 +659,42 @@ func setCompressorMode(val string) []string {
 	}
 }
 
-func printThroughput(requestCount uint64, requestSize int, responseCount uint64, responseSize int, benchTime time.Duration) {
-	requestThroughput := float64(requestCount) * float64(requestSize) * 8 / benchTime.Seconds()
-	responseThroughput := float64(responseCount) * float64(responseSize) * 8 / benchTime.Seconds()
-	fmt.Printf("Number of requests:  %v\tRequest throughput:  %v bit/s\n", requestCount, requestThroughput)
-	fmt.Printf("Number of responses: %v\tResponse throughput: %v bit/s\n", responseCount, responseThroughput)
-	fmt.Println()
-}
-
 func main() {
 	opts := processFlags()
 	before(opts)
+
 	s := stats.NewStats(numStatsBuckets)
-	s.SortLatency()
-	var memStats runtime.MemStats
-	var results testing.BenchmarkResult
-	var startAllocs, startBytes uint64
-	var startTime time.Time
-	var startTimer = func() {
-		runtime.ReadMemStats(&memStats)
-		startAllocs = memStats.Mallocs
-		startBytes = memStats.TotalAlloc
-		startTime = time.Now()
-	}
-	var stopTimer = func(count uint64) {
-		runtime.ReadMemStats(&memStats)
-		results = testing.BenchmarkResult{
-			N:         int(count),
-			T:         time.Since(startTime),
-			Bytes:     0,
-			MemAllocs: memStats.Mallocs - startAllocs,
-			MemBytes:  memStats.TotalAlloc - startBytes,
+	featuresNum := makeFeaturesNum(opts)
+	sf := sharedFeatures(featuresNum)
+	mode := ""
+
+	var (
+		makeStart = func(mode string, bf stats.Features) func() {
+			return func() { s.StartRun(mode, bf, sf) }
```

Comment:
Everywhere you use start (in unconstrainedStreamBenchmark and in runBenchmark) , you have access to bf. Can we make the type `startFunc` to be `func(stats.Features)` and use it as `start(bf)`? So, makeStart will be 
```golang
makeStart = func(mode string, bf stats.Features) startFunc {
  return func(bf stats.Features) { s.StartRun(mode, bf, sf) }
}
```

Reply:
For me, an even better approach will be the following.
Make the type `startFunc` to be `func(string, stats.Features)`
define `start` as
```golang
start = func(mode string, bf stats.Features) {s.StartRun(mode, bf, sf)}
```

in `unconstrainedStreamBenchmark`, when calling `start` `mode` will always be `workloadsUnconstrained`

`runBenchmark` will get a `mode` parameter from `unaryBenchmark` and from `streamBenchmark` and will call `start` with that parameter.

Reply:
Done. Thanks.